javascript/modules/client/src/client-erc20.ts

```diff
@@ -42,51 +49,129 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
 
   /** Contains all the generic high level methods to interact with a DAO */
   methods = {
-    createProposal: (params: ICreateProposalParams) =>
+    /**
+     * Creates a new proposal on the given ERC20 plugin contract
+     *
+     * @param {ICreateProposalParams} params
+     * @return {*}  {AsyncGenerator<ProposalCreationStepValue>}
+     * @memberof ClientErc20
+     */
+    createProposal: (params: ICreateProposalParams): AsyncGenerator<ProposalCreationStepValue> =>
       this._createProposal(params),
-    voteProposal: (proposalId: string, vote: VoteOptions) =>
+    /**
+     * Cast a vote on the given proposal using the client's wallet. Depending on the proposal settings, an affirmative vote may execute the proposal's actions on the DAO.
+     *
+     * @param {string} proposalId
+     * @param {VoteOptions} vote
+     * @return {*}  {AsyncGenerator<VoteProposalStepValue>}
+     * @memberof ClientErc20
+     */
+    voteProposal: (proposalId: string, vote: VoteOptions): AsyncGenerator<VoteProposalStepValue> =>
       this._voteProposal(proposalId, vote),
-    executeProposal: (proposalId: string) => this._executeProposal(proposalId),
-    // setDaoConfig: (address: string, config: DaoConfig) =>
-    //   this._setDaoConfig(address, config),
-    // setVotingConfig: (address: string, config: VotingConfig) =>
-    //   this._setVotingConfig(address, config),
-    /** Retrieves the list of Ethereum addresses that can cast weighted votes on the given DAO's proposals */
-    getMembers: (daoAddressOrEns: string) => this._getMembers(daoAddressOrEns),
-    /** Retrieves the list of proposals of the given DAO */
-    getProposals: (daoAddressOrEns: string) =>
-      this._getProposals(daoAddressOrEns),
+    /**
+     * Executes the given proposal, provided that it has already passed
+     *
+     * @param {string} proposalId
+     * @return {*}  {AsyncGenerator<ExecuteProposalStepValue>}
+     * @memberof ClientErc20
+     */
+    executeProposal: (proposalId: string): AsyncGenerator<ExecuteProposalStepValue> =>
+      this._executeProposal(proposalId),
+    /**
+     * Sets the voting configuration in a erc20 proposal given a proposalId and a configuration
+     *
+     * @param {VotingConfig} config
+     * @return {*}  {AsyncGenerator<SetVotingConfigStepValue>}
+     * @memberof ClientErc20
+     */
+    setPluginConfig: (config: VotingConfig): AsyncGenerator<SetVotingConfigStepValue> =>
+      this._setPluginConfig(config),
+
```

Comment:
This should disappear and instead be an `encoder` when creating a proposal. (We should have only one correct way of doing things)

Reply:
Not sure why, this is for changing the plugin configuration, this should be part of the plugin client